File: 0318/SeoyeonHong/28353 고양이 카페.py
N, K = map(int, input().split())

weights = list(map(int, input().split()))
weights.sort() # 오름차순 정렬
light = 0 # 가벼운 고양이 인덱스
heavy = N-1 # 무거운 고양이 인덱스
happy = 0 # 행복한 사람 수
while light < heavy:
    if weights[light] + weights[heavy] <= K: # 두 고양이의 무게의 합이 K 이하라면
        happy += 1
        light += 1
        heavy -= 1
    else:
        heavy -= 1
    
print(happy)

File: 0318/SoohyunKim/28353_고양이카페.js
const input = require('fs').readFileSync('/dev/stdin').toString().trim().split('\n');
const [N, K, ...cats] = input.flatMap(s => s.split(' ').map(Number));

cats.sort((a, b) => a - b);

let left = 0;
let right = N - 1;
let answer = 0;

while (left < right) {
  if(cats[left] + cats[right] <= K) {
    answer++;
    left++;
    right--;
  } else {
    right--;
  }
}

console.log(answer);